tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/field/default/field_default_model_test.dart

```diff
@@ -0,0 +1,155 @@
+import 'package:serverpod_cli/analyzer.dart';
+import 'package:serverpod_cli/src/analyzer/code_analysis_collector.dart';
+import 'package:serverpod_cli/src/analyzer/models/stateful_analyzer.dart';
+import 'package:serverpod_cli/src/generator/code_generation_collector.dart';
+import 'package:serverpod_cli/src/test_util/builders/generator_config_builder.dart';
+import 'package:serverpod_cli/src/test_util/builders/model_source_builder.dart';
+import 'package:test/test.dart';
+
+void main() {
+  var config = GeneratorConfigBuilder().build();
+
+  group('Given a class with fields with a "defaultModel" keyword', () {
+    test(
+      'when the field is of type DateTime and the default is set to "now", then the field should have a "default model" value',
+      () {
+        var models = [
+          ModelSourceBuilder().withYaml(
+            '''
+            class: Example
+            table: example
+            fields:
+              dateTimeType: DateTime, defaultModel=now
+            ''',
+          ).build()
+        ];
+
+        var collector = CodeGenerationCollector();
+        var definitions =
+            StatefulAnalyzer(config, models, onErrorsCollector(collector))
+                .validateAll();
+
+        expect(collector.errors, isEmpty);
+
+        var definition = definitions.first as ClassDefinition;
+        expect(definition.fields.last.defaultModelValue, 'now');
+      },
+    );
```

Comment:
Suggestion: Do a test with uppercase chars as well, how do we handle 'NOW'? is it important?